### ISSUE TITLE
Crash when search thread running

### ISSUE THREAD

**Issue by StarkZhidian** [Rosemoe/sora-editor#566]
stack trace：

```

Exception java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
  at io.github.rosemoe.sora.widget.EditorSearcher$SearchRunnable.run (EditorSearcher.java:23)
  at java.lang.Thread.run (Thread.java:1012)

```

**Comment by StarkZhidian** [comment]
From stack trace, it like occurred at here:
![image](https://github.com/Rosemoe/sora-editor/assets/27915437/19f9b618-dc00-41cb-b579-5489cecad043)

Maybe should judge the search status before the search thread actually run

**Comment by Rosemoe** [comment]
`searchOptions` should be changed to `options` here. I wrote it wrongly.

**Comment by Rosemoe** [comment]
fix by #568
